src/Arcus.WebApi.Unit/Security/Authentication/CertificateAuthenticationFilterTests.cs

```diff
@@ -0,0 +1,230 @@
+﻿using System;
+using System.Net;
+using System.Net.Http;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+using Arcus.Security.Secrets.Core.Interfaces;
+using Arcus.WebApi.Security.Authentication;
+using Arcus.WebApi.Unit.Hosting;
+using Arcus.WebApi.Unit.Security.Doubles;
+using Xunit;
+
+namespace Arcus.WebApi.Unit.Security.Authentication
+{
+    public class CertificateAuthenticationFilterTests : IDisposable
+    {
+        private readonly TestApiServer _testServer = new TestApiServer();
+
+        [Fact]
+        public async Task AuthorizedRoute_WithCertificateAuthentication_ShouldFailWithUnauthorized_WhenClientCertificateSubjectNameDoesntMatch()
+        {
+            // Arrange
+            string subjectKey = "subject", subjectValue = $"subject-{Guid.NewGuid()}";
+            _testServer.AddService<ISecretProvider>(new InMemorySecretProvider((subjectKey, subjectValue)));
+            _testServer.AddService(
```

Comment:
Just a general remark for all tests, if you register CertificateAuthenticationValidator seperately from CertificateAuthenticationConfig you can avoid calling it like this and switch to :
```csharp
 _testServer.AddService(CertificateAuthenticationConfig()
                            .WithSubject(X509ValidationLocation.SecretProvider, subjectKey));
```

And move the registration of `CertificateAuthenticationValidator` to the test server

Reply:
Just a suggestion to reduce duplication

Reply:
> And move the registration of CertificateAuthenticationValidator to the test server

That would not be a good idea if we want to reuse a clean-slate test server for our tests.

The registration  of the validator is not the only duplication that exists because of the 'simple'-testcase setup we must use and if we would follow allong this 'basic-approach' it would make sense to explicitly register it directly in the test case and not somewhere hidden.

Reply:
I don't really see a problem why we could not do it but is ok, was just a suggestion.

Resolving!